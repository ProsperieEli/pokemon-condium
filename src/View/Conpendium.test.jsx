import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Controls from "../Components/Control/Control";
import PokemonList from "../Components/PokemonList/PokemonList";

it("it should render the Pokemon", () => {
  const pokemons = [
    {
      abilityOne: "Thick-fat",
      abilityTwo: "Vine-Whip",
      generation: "-MEGA-",
      id: 4,
      name: "Venusaur-mega",
      pic: "https://assets.pokemon.com/assets/cms2/img/pokedex/full/003_f2.png",
      typeOne: "Grass",
      typeTwo: "Poison",
    },
  ];
  const { container } = render(<PokemonList pokemons={pokemons} />);

  expect(container).toMatchSnapshot();
});

it("it should render the name, type search", () => {
  const { container } = render(
    <Controls name="Pikachu" types={["electric"]} selectedType="electric" />
  );

  expect(container).toMatchSnapshot();
});
