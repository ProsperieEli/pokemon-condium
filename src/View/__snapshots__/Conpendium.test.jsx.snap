// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the Pokemon 1`] = `
<div>
  <ul
    aria-label="pokemonList"
    class="pokemonList"
  >
    <li
      class="listItem"
    >
      <figure
        aria-label="pokemon"
      >
        <img
          alt="Venusaur-mega--MEGA-"
          src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/003_f2.png"
        />
        <div
          class="details"
        >
          <h2>
            Venusaur-mega
          </h2>
          <p>
            -MEGA-
          </p>
          <p>
            Thick-fat
            /Vine-Whip
          </p>
          <p>
            Grass
            /Poison
          </p>
        </div>
      </figure>
    </li>
  </ul>
</div>
`;

exports[`it should render the name, type search 1`] = `
<div>
  <div
    class="controlsContainer"
  >
    <form
      aria-label="searchForm"
    >
      <h3>
        Search a Pokemon
      </h3>
      <label>
        Name:
        <input
          id="pokename"
          name="name"
          type="text"
          value="Pikachu"
        />
      </label>
      <button
        type="submit"
      >
        PokeSearch!
      </button>
    </form>
    <div>
      <h3>
        Filter Pokemon
      </h3>
      <select>
        <option
          value="all"
        >
          ALL
        </option>
        <option />
      </select>
    </div>
  </div>
</div>
`;
